CMakeLists.txt

```diff
@@ -220,12 +237,12 @@ if(CMAKE_C_COMPILER_ID STREQUAL "Clang")
     -Wno-tautological-compare
     )
 else() # GCC assumed
-  zephyr_cc_option(
-    -Wno-unused-but-set-variable
-    -fno-reorder-functions
-    )
   if(NOT ${ZEPHYR_TOOLCHAIN_VARIANT} STREQUAL "xcc")
-    zephyr_cc_option(-fno-defer-pop)
+    zephyr_cc_option(
+      -Wno-unused-but-set-variable
+      -fno-reorder-functions
+      -fno-defer-pop
+    )
```

Comment:
It should be possible to omit the if statement.

Reply:
ok